### PR TITLE
Fix container restore lint and unit tests

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind ci

#### What this PR does / why we need it:
Fix the lint report by inverting the logic in the loop.
#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by kwilczynski** [comment]
Let's get this in as soon as possible, as I was about to open Pull Requests with backports of the fix this code belongs to.

**Comment by saschagrunert** [comment]
> Let's get this in as soon as possible, as I was about to open Pull Requests with backports of the fix this code belongs to.

Unit tests are broken as well, working on it now :+1:

**Comment by kwilczynski** [comment]
> > Let's get this in as soon as possible, as I was about to open Pull Requests with backports of the fix this code belongs to.
> 
> Unit tests are broken as well, working on it now 👍

Thank you! Appreciated. :bow:

**Comment by saschagrunert** [comment]
CI looks now green'ish

File: server/container_restore.go [hunk 1]
```diff
@@ -287,14 +287,17 @@ func (s *Server) CRImportCheckpoint(
...
+			if createMount.ContainerPath != m.Destination {
+				continue
 			}
+
+			mount.HostPath = createMount.HostPath
+			mount.Readonly = createMount.Readonly
+			mount.RecursiveReadOnly = createMount.RecursiveReadOnly
+			mount.Propagation = createMount.Propagation
+			mount.RecursiveReadOnly = createMount.RecursiveReadOnly
+			bindMountFound = true
```
[the remark this answers is not in the document]
Comment (saschagrunert) [reply server/container_restore.go:299]: @kwilczynski that's where the linter complained: 

```
  Error: server/container_restore.go:290:4: nestingReduce: invert if cond, replace body with `continue`, move old body after the statement (gocritic)
  			if createMount.ContainerPath == m.Destination {
  			^
  
  Error: issues found
```

[the remark this answers is not in the document]
Comment (kwilczynski) [reply server/container_restore.go:299]: @saschagrunert, I see... while I do not agree with the linter here, LOL, it's fine to change it then to make it pass.

The extra assignment needs to be removed, though.

[the remark this answers is not in the document]
Comment (saschagrunert) [reply server/container_restore.go]: True, I removed that.

**Pull request merged**